client/post-editor/media-modal/test/index.jsx

```diff
@@ -85,194 +76,229 @@ describe( 'EditorMediaModal', () => {
-		tree.deleteMedia();
-
-		return new Promise( ( resolve ) => {
-			process.nextTick( function () {
-				expect( deleteMedia ).toHaveBeenCalledWith(
-					DUMMY_SITE.ID,
-					DUMMY_MEDIA.map( ( { ID } ) => ID )
-				);
-				resolve();
-			} );
+	test( 'should prompt to delete multiple items from the list view', async () => {
+		const user = userEvent.setup();
+
+		MediaLibrary.mockImplementationOnce( ( { onDeleteItem } ) => {
+			return <button onClick={ onDeleteItem }>Delete</button>;
 		} );
+
+		const props = { ...baseProps, selectedItems: DUMMY_MEDIA };
+		const { container } = render( <EditorMediaModal { ...props } /> );
+
+		Modal.setAppElement( container );
+
+		await user.click( screen.getByRole( 'button', { name: 'Delete' } ) );
+
+		expect( baseProps.deleteMedia ).toHaveBeenCalledWith(
+			DUMMY_SITE.ID,
+			DUMMY_MEDIA.map( ( { ID } ) => ID )
+		);
 	} );
 
 	test( 'should show no buttons if editing an image', () => {
-		const tree = shallow(
-			<EditorMediaModal
-				{ ...baseProps }
-				selectedItems={ [] }
-				view={ ModalViews.IMAGE_EDITOR }
-				setView={ spy }
-			/>
-		).instance();
+		const props = {
+			...baseProps,
+			selectedItems: [],
+			view: ModalViews.IMAGE_EDITOR,
+			setView: spy,
+		};
+		const { container } = render( <EditorMediaModal { ...props } /> );
 
-		const buttons = tree.getModalButtons();
+		Modal.setAppElement( container );
 
-		expect( buttons ).toBeUndefined();
+		expect( screen.getByTestId( 'image-editor' ) ).toBeVisible();
+		expect( screen.queryAllByRole( 'button' ) ).toHaveLength( 0 );
 	} );
 
 	test( 'should show a Copy to media library button when viewing external media (no selection)', () => {
-		const tree = shallow(
+		const { container } = render(
 			<EditorMediaModal
 				{ ...baseProps }
 				view={ ModalViews.DETAIL }
 				setView={ spy }
 				selectedItems={ [] }
+				source="external"
 			/>
-		).instance();
+		);
 
-		tree.setState( { source: 'external' } );
-		const buttons = tree.getModalButtons();
+		Modal.setAppElement( container );
 
-		expect( buttons.length ).toEqual( 2 );
-		expect( buttons[ 1 ].label ).toEqual( 'Copy to media library' );
+		expect( screen.getByTestId( 'media-modal-detail-base' ) ).toBeVisible();
+		expect( screen.getByRole( 'button', { name: /copy to media library/i } ) ).toBeVisible();
 	} );
 
 	test( 'should show a Copy to media library button when 1 external image is selected', () => {
-		const tree = shallow(
+		const { container } = render(
 			<EditorMediaModal
 				{ ...baseProps }
 				view={ ModalViews.DETAIL }
 				selectedItems={ DUMMY_MEDIA.slice( 0, 1 ) }
 				setView={ spy }
+				source="external"
 			/>
-		).instance();
+		);
 
-		tree.setState( { source: 'external' } );
-		const buttons = tree.getModalButtons();
+		Modal.setAppElement( container );
 
-		expect( buttons.length ).toEqual( 2 );
-		expect( buttons[ 1 ].label ).toEqual( 'Copy to media library' );
+		expect( screen.queryAllByRole( 'button' ) ).toHaveLength( 2 );
```

Comment:
When querying for existing elements, it's preferred to use `get*` queries rather than `query` queries:

```suggestion
		expect( screen.getAllByRole( 'button' ) ).toHaveLength( 2 );
```